### PR TITLE
URN Refactor

### PR DESCRIPTION
**Pull request by jhrr**
- Refactors throughout the project to complete using the new `URN` utility everywhere.
- Expands the functionality of the `URN` utility.
- Eliminates the noisier parts of URN munging in `CTSImporter` and tests, relocating that labour as much as possible in the `URN` class itself.
- Unittest and fuzztest the `URN` class.
- Refactor the test suite for `CTSImporter`.
- Add `coverage` to the project.

### CODE REVIEW

File: sv_mini_atlas/library/urn.py [hunk 1]
```diff
@@ -84,19 +103,24 @@ def to_no_passage(self):
...
         attr_name = f"to_{label}"
         try:
             value = getattr(self, attr_name)
         except TypeError:
-            raise ValueError(f'URN has no "{label}" component')
+            raise ValueError(f"URN has no component: {label}")
 
-        if key == self.NO_PASSAGE and self.parsed["work"]:
-            return value
+        # TODO: Is there any time we don't want a trailing ":"?
```
Comment (jhrr) [inline sv_mini_atlas/library/urn.py]: @jacobwegner this is my only outstanding 'not sure'.

File: sv_mini_atlas/library/urn.py [hunk 2]
```diff
@@ -84,19 +103,25 @@ def to_no_passage(self):
...
         attr_name = f"to_{label}"
         try:
             value = getattr(self, attr_name)
         except TypeError:
-            raise ValueError(f'URN has no "{label}" component')
+            raise ValueError(f"URN has no component: {label}")
 
-        if key == self.NO_PASSAGE and self.parsed["work"]:
-            return value
+        # TODO: Is there any time we don't want a trailing ":" for any URN
```
Comment (jhrr) [inline sv_mini_atlas/library/urn.py]: @jacobwegner this is my only outstanding 'not sure'.

Comment (jacobwegner) [reply sv_mini_atlas/library/urn.py]: @jhrr:

I think we always want the trailing `:` on any work-component level URN.

Once we have a passage component leave URN, we never have the trailing colon.

Does that help to clarify?

Comment (jhrr) [reply sv_mini_atlas/library/urn.py]: @jacobwegner yep that was also my thinking, so I think the [lines I commented](https://github.com/scaife-viewer/sv-mini-atlas/pull/3/files#diff-f059ee644c60f00cb5173d1aefb52c96R123-R124) can be removed altogether?

Put differently, I think we'll always be appending `:` if we use `up_to`?

Comment (jacobwegner) [reply sv_mini_atlas/library/urn.py]: Agreed; this might have been cruft from the earlier backport I did.  Remove at will!

**Review by jacobwegner — APPROVED** [review]

**Comment by jhrr** [comment]
very good, thanks!

**Pull request merged**